src/core/server/http/lifecycle/auth.ts

```diff
@@ -38,20 +39,34 @@ export interface Authenticated extends AuthResultParams {
 }
 
 /** @public */
-export type AuthResult = Authenticated;
+export interface AuthNotHandled extends AuthNotHandledResultParams {
+  type: AuthResultType.notHandled;
+}
+
+/** @public */
+export type AuthResult = Authenticated | AuthNotHandled;
 
 const authResult = {
-  authenticated(data: Partial<AuthResultParams> = {}): AuthResult {
+  authenticated(data: AuthResultParams = {}): AuthResult {
     return {
       type: AuthResultType.authenticated,
       state: data.state,
       requestHeaders: data.requestHeaders,
       responseHeaders: data.responseHeaders,
     };
   },
+  notHandled(data: AuthNotHandledResultParams = {}): AuthResult {
+    return {
+      type: AuthResultType.notHandled,
+      responseHeaders: data.responseHeaders,
+    };
+  },
   isAuthenticated(result: AuthResult): result is Authenticated {
     return result && result.type === AuthResultType.authenticated;
   },
+  isNotHandled(result: AuthResult): result is AuthNotHandled {
+    return result && result.type === AuthResultType.notHandled;
+  },
```

Comment:
Are the `result &&` really needed? It seems `result` should not be false-ish with current typing?

Reply:
right. okay I will update to `Boolean(result.?type === ...)`